src/Core/src/Core/IImage.cs

```diff
@@ -0,0 +1,31 @@
+﻿#nullable enable
+
+using System;
+
+namespace Microsoft.Maui
+{
+	public interface IImage : IView, IImageSourcePart
+	{
+		Aspect Aspect { get; }
+
+		bool IsOpaque { get; }
+	}
+
+	public interface IImageSourcePart
+	{
+		IImageSource? Source { get; }
+
+		bool IsAnimationPlaying { get; }
+
+		void UpdateIsLoading(bool isLoading);
+	}
+
+	public interface IImageSourcePartEvents
```

Comment:
No idea what to call these... Naming is hard.

Reply:
There are 3 parts:
1. the main control (Image, Button, ImageButton, ToolbarItem)
2. the image source properties
3. the events (most cases this will be the same thing)

Right now we have `IImage : IImageSourcePart`, but we will have `IButton :  IImageSourcePart`.

Not sure what props should be where though.

Reply:
Should IImageSourcePart implement this interface?

I don't see anything implementing IImageSourcePartEvents

Reply:
Not necessarily. Not sure. It could be the case that it is even a separate object... Who knows... I need to implement the button and maybe the page things and see where it fits.